src/kernel/capinv.c

```diff
@@ -370,8 +372,7 @@ cap_cpy(struct captbl *t, capid_t cap_to, capid_t capin_to, capid_t cap_from, ca
 		old_v = *f;
 
 		/* Cannot copy frame, or kernel entry. */
-		if ((old_v & PGTBL_COSFRAME) || !(old_v & PGTBL_USER)) return -EPERM;
-		/* TODO: validate the type is appropriate given the value of *flags */
+		if (chal_pgtbl_flag_exist(old_v, PGTBL_COSFRAME) || !chal_pgtbl_flag_exist(old_v, PGTBL_USER)) return -EPERM;
```

Comment:
Just wondering, because you have a API for flag operations, wouldn't it be better to have `chal_pgtbl_flag_notexist` kind instead of `!xxxx_exist` api?